### PR TITLE
Bugfix/381/failed when the job name was null or not found

### LINKED ISSUE #381: [Bug] [zos_job_query] failed when the `job_name` was null or not found
`- name: Check CTL Up
        zos_job_query:
          job_name: "{{ DFS_IMS_SSID }}CTL"
        register: ctl_output`

`TypeError: Value of unknown type: <class 'RuntimeError'>, List FAILED! no such job was found.
fatal: [zosendpoint]: FAILED! => {
    "changed": false,
    "module_stderr": "Shared connection to api.apps.z-stack-int.cp.fyre.ibm.com closed.\r\n",
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_zos_job_query_payload_8yzfa7vq/ansible_zos_job_query_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_job_query.py\", line 165, in run_module\r\n  File \"/tmp/ansible_zos_job_query_payload_8yzfa7vq/ansible_zos_job_query_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_job_query.py\", line 211, in query_jobs\r\nRuntimeError: List FAILED! no such job was found.\r\n\r\nDuring handling of the above exception, another exception occurred:\r\n\r\nTraceback (most recent call last):\r\n  File \"/u/ibmuser/.ansible/tmp/ansible-tmp-1660311444.913656-9146-196003178089563/AnsiballZ_zos_job_query.py\", line 107, in <module>\r\n    _ansiballz_main()\r\n  File \"/u/ibmuser/.ansible/tmp/ansible-tmp-1660311444.913656-9146-196003178089563/AnsiballZ_zos_job_query.py\", line 99, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/u/ibmuser/.ansible/tmp/ansible-tmp-1660311444.913656-9146-196003178089563/AnsiballZ_zos_job_query.py\", line 47, in invoke_module\r\n    runpy.run_module(mod_name='ansible_collections.ibm.ibm_zos_core.plugins.modules.zos_job_query', init_globals=dict(_module_fqn='ansible_collections.ibm.ibm_zos_core.plugins.modules.zos_job_query', _modlib_path=modlib_path),\r\n  File \"/usr/lpp/IBM/cyp/v3r9/pyz/lib/python3.9/runpy.py\", line 210, in run_module\r\n    return _run_module_code(code, init_globals, run_name, mod_spec)\r\n  File \"/usr/lpp/IBM/cyp/v3r9/pyz/lib/python3.9/runpy.py\", line 97, in _run_module_code\r\n    _run_code(code, mod_globals, init_globals,\r\n  File \"/usr/lpp/IBM/cyp/v3r9/pyz/lib/python3.9/runpy.py\", line 87, in _run_code\r\n    exec(code, run_globals)\r\n  File \"/tmp/ansible_zos_job_query_payload_8yzfa7vq/ansible_zos_job_query_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_job_query.py\", line 261, in <module>\r\n  File \"/tmp/ansible_zos_job_query_payload_8yzfa7vq/ansible_zos_job_query_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_job_query.py\", line 257, in main\r\n  File \"/tmp/ansible_zos_job_query_payload_8yzfa7vq/ansible_zos_job_query_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_job_query.py\", line 168, in run_module\r\n  File \"/tmp/ansible_zos_job_query_payload_8yzfa7vq/ansible_zos_job_query_payload.zip/ansible/module_utils/basic.py\", line 1533, in fail_json\r\n  File \"/tmp/ansible_zos_job_query_payload_8yzfa7vq/ansible_zos_job_query_payload.zip/ansible/module_utils/basic.py\", line 1506, in _return_formatted\r\n  File \"/tmp/ansible_zos_job_query_payload_8yzfa7vq/ansible_zos_job_query_payload.zip/ansible/module_utils/common/parameters.py\", line 887, in remove_values\r\n  File \"/tmp/ansible_zos_job_query_payload_8yzfa7vq/ansible_zos_job_query_payload.zip/ansible/module_utils/common/parameters.py\", line 461, in _remove_values_conditions\r\nTypeError: Value of unknown type: <class 'RuntimeError'>, List FAILED! no such job was found.\r\n",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1`

### PR DESCRIPTION
**Pull request by AndreMarcel99**
##### SUMMARY
Add a variable to change the value depends of the parameters to jo_id, job_name or both.

Fixed #381 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
Add variable name to change value depends on names.

##### ADDITIONAL INFORMATION
Change test where some values to support these Bugfix.
The logic is if the job_id and job_name are not given the message will be the _job_name with the job_id i_
In other case will verify will is given and just use as the job not founded.

Captura de pantalla 2023-04-17 a la(s) 13 25 36
(Ignore the change in encode of utils, already in dev)

### CODE REVIEW

**Comment by ddimatos** [comment]
Note that is a 1.7 work item, thus wait till we branch `dev` before trying to merge.

File: plugins/module_utils/job.py
Comment (fernandofloresg) [inline plugins/module_utils/job.py:1]: Correct me if I'm wrong but if job_id = 15 and job_name = '*', then return message would be "The job 15 could not be found.".
In a similar way, if job_id = '*' and job_name = "HELLOPGM", then return message would be "The job HELLOPGM could not be found."

I suggest two things:
1. Change the variable name to use a name that indicates the use of it i.e. job_not_found_msg, not_found_msg, ...
2. Change the job_not_founded message for those two cases above. Would rather have something like "The job with id 15 could not be found." and "The job with name HELLOPGM could not be found."

**Review by fernandofloresg — CHANGES_REQUESTED** [review]
Comments are in each file.

File: plugins/module_utils/encode.py
Comment (AndreMarcel99) [reply plugins/module_utils/encode.py:1]: It's in dev like that [here](https://github.com/ansible-collections/ibm_zos_core/blob/a856d18e9797f1d2eacf4ca7ab350b9f4cc3b390/plugins/module_utils/encode.py#L456) and [here](https://github.com/ansible-collections/ibm_zos_core/blob/a856d18e9797f1d2eacf4ca7ab350b9f4cc3b390/plugins/module_utils/encode.py#L499)

File: plugins/module_utils/job.py
Comment (AndreMarcel99) [reply plugins/module_utils/job.py:1]: I will work with the second suggest, but about the first one, steps back in the code already take care of job_name='' or job_id='' put it a * as value [here](https://github.com/ansible-collections/ibm_zos_core/blob/336a7980e35951740e44055b6910252e33f4ef42/plugins/module_utils/job.py#L59)

File: plugins/module_utils/encode.py
Comment (fernandofloresg) [inline plugins/module_utils/encode.py:1]: If there are not changes to this file could you just pull the branch dev version of it for this to stop appearing in the PR ? I'm afraid this would add the function a third time.

File: changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml
Comment (fernandofloresg) [inline changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml:2]: Please change the word "specify" to "specifies".

**Review by fernandofloresg — CHANGES_REQUESTED** [review]
Overall looks good, just update those two details. zos_encode shouldn't appear in the PR at all.

File: plugins/module_utils/encode.py
Comment (fernandofloresg) [inline plugins/module_utils/encode.py:1]: Caution! This will remove the file from dev... encode file shouldn't even appear in this PR if is identical to the one already in dev. Kindly checkout the latest from there.

File: changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml
Comment (fernandofloresg) [inline changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml:2]: I'm sorry I did not saw the first sentence. This should be worded like :
"Error message did not specify the job not found. Fix now specifies the job_id or job_name searching for, to ensure more information to the user."

**Review by fernandofloresg — CHANGES_REQUESTED** [review]

File: plugins/module_utils/encode.py
Comment (AndreMarcel99) [reply plugins/module_utils/encode.py:1]: Is it solved?

Comment (fernandofloresg) [reply plugins/module_utils/encode.py:1]: No, it still appears as before. Let me try something...

**Comment by fernandofloresg** [comment]
@AndreMarcel99 That should be it. Had some issues going back and forth but overall issue was : 
1. Merge from dev
2. use git command `git checkout dev ./plugins/module_utils/encode.py` to checkout the file from the requested branch into the current branch
3. use git command `it diff bugfix/381/failed_when_the_job_name_was_null_or_not_found:plugins/module_utils/encode.py dev:plugins/module_utils/encode.py`

File: changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml
Comment (fernandofloresg) [inline changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml:1]: Sorry for still correcting this with new comments but just noticed. 

Correct tag is bugfixes not bugfix(even though is one). And GH is crying over the file not having a new line at the end kindly add it.

**Review by fernandofloresg — CHANGES_REQUESTED** [review]
Thank you so much for going through all the changes. Was a good exercise, only thing pending now is to run the pipeline and the comment about the change log.

File: changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml
Comment (rexemin) [reply changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml:1]: GitHub doesn't require the file to have a new line at the end. The symbol it's just there to indicate that it is there. Couldn't find in the style guidelines if fragments require newlines, so I think @AndreMarcel99 can leave the file as is.

Comment (fernandofloresg) [reply changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml:1]: Thanks for clarification @rexemin

**Comment by AndreMarcel99** [comment]
Captura de pantalla 2023-04-25 a la(s) 12 02 27

With the branch update

**Review by fernandofloresg — APPROVED** [review]
Merged the latest from dev, was only doc changes no need to run the pipeline again. LGTM

**Review by rexemin — APPROVED** [review]
Looks good to me.

File: changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml
Comment (ddimatos) [reply changelogs/fragments/747-failed_when_the_job_name_was_null_or_not_found.yaml:1]: I have noticed that too, we seem to build without it just fine, wish GH would not make it look so severe :(

File: plugins/module_utils/job.py
Comment (ddimatos) [reply plugins/module_utils/job.py:1]: BTW, i think (recalling from memory) , the reason fro the asterisk (`*`) is we can't query on None , we need to pass something to our query and we assume if there is no specifics we just go after everything. Speaking about this line of code `job_id = parsed_args.get("job_id") or "*"`

File: plugins/module_utils/encode.py
Comment (ddimatos) [reply plugins/module_utils/encode.py:1]: Looks like this is resolved, i checkout the dev branch and searched this feature branch and could not find a duplicate `uss_tag_encoding`, might be helpful in the future for adhoc items like this add a PR or issue number to follow.

**Comment by ddimatos** [comment]
See full regression `/job/zosAnsible/1721/console` 
image

**Pull request merged**
